From 7ded2f72e73e5ef4f4380785c081346de5981c0d Mon Sep 17 00:00:00 2001
From: Alex Stewart <alex.stewart@ni.com>
Date: Wed, 12 May 2021 08:05:05 -0500
Subject: [PATCH] CMakeLists: fixup RPATH

When cross-compiling, cmake writes RPATHs using the host machine's
paths. Normally, when cmake executes the `install` target, it rewrites
the paths to conform to the install prefix. Because the current cmake
implementation does not express a real `install` target, the paths are
never fixed.

Set BUILD_WITH_INSTALL_PATH=TRUE for the ni_grpc_device_server target,
so that cross compile binaries always use the supposed install prefix
paths.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6cad5ef..b187fa4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -189,6 +189,8 @@ target_link_libraries(ni_grpc_device_server
    nlohmann_json::nlohmann_json
    )
 
+set_target_properties(ni_grpc_device_server PROPERTIES BUILD_WITH_INSTALL_RPATH TRUE)
+
 if(WIN32)
   target_link_libraries(ni_grpc_device_server Delayimp niScope niswitch nisync nidcpower)
   set_target_properties(ni_grpc_device_server PROPERTIES LINK_FLAGS "/DELAYLOAD:niScope.dll /DELAYLOAD:niswitch.dll /DELAYLOAD:nisync.dll /DELAYLOAD:nidcpower.dll")
